sklearn/base.py

```diff
@@ -266,6 +266,7 @@ def __repr__(self):
 ###############################################################################
 class ClassifierMixin(object):
     """Mixin class for all classifiers in scikit-learn."""
+    estimator_type = "classifier"
```

Comment:
> But the point of this tag system is to let third-party developers implement
> classes without inheriting from our base classes. Private variables can be
> changed.

> On the other hand, a third-party developer doesn't want his code to
> break because we changed variable names. We need to make this tag part
> of our API and commit to it.

Agreed, but it's going to make tab completion ugly, and it's going to
worry/confuse our users.

In a sense, this is the same thing as an __add__ function, which is a
private variable. So in Python land, typing-related class information are
usually coded why private attributes, I would say.